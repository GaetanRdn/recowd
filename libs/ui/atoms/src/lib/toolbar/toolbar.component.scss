:host {
  display: flex;
  flex-flow: row;
  align-items: center;
  gap: 0.4rem;
  justify-content: flex-start;
  height: 6rem;
  padding: 0 1.6rem;
}

@mixin rc-toolbar($config) {
  $typography: map-get(map-get($config, typography), toolbar);
  $colors: map-get($config, colors);
  $primary: map-get($colors, primary);

  rc-toolbar {
    @each $key in font-size, font-family, font-weight, line-height {
      #{$key}: map-get($typography, $key);
    }

    @each $key in color, background-color {
      #{$key}: map-get($primary, $key);
    }
  }
}
