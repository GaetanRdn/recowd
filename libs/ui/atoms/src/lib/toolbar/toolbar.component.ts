import { ChangeDetectionStrategy, Component, NgModule } from '@angular/core';

@Component({
  selector: 'rc-toolbar',
  template: ` <ng-content></ng-content>`,
  styleUrls: ['./toolbar.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ToolbarComponent {}

@NgModule({
  declarations: [ToolbarComponent],
  exports: [ToolbarComponent],
})
export class ToolbarModule {}
