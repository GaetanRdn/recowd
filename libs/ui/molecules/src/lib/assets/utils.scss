@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
