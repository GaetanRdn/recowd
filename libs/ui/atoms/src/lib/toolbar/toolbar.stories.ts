import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { MaterialIconComponent } from '../material-icon/material-icon.component';
import { ToolbarComponent } from './toolbar.component';
import { ButtonModule } from '../button/button.directive';
import { BADGE } from '@geometricpanda/storybook-addon-badges';

export default {
  title: 'atoms/toolbar',
  component: ToolbarComponent,
  parameters: {
    badges: [BADGE.STABLE],
    jest: 'toolbar.directive',
  },
  decorators: [
    moduleMetadata({
      imports: [ToolbarComponent, MaterialIconComponent, ButtonModule],
    }),
  ],
} as Meta<ToolbarComponent>;

const templateForOnlyText: Story<ToolbarComponent> = () => ({
  template: `<rc-toolbar>Design System</rc-toolbar>`,
});

export const OnlyText = templateForOnlyText.bind({});

const templateForOnlyButton: Story<ToolbarComponent> = () => ({
  template: `<rc-toolbar><button rcButtonIcon><rc-material-icon>menu</rc-material-icon></button></rc-toolbar>`,
});

export const OnlyButton = templateForOnlyButton.bind({});

const templateForBothTextAndButton: Story<ToolbarComponent> = () => ({
  template: `<rc-toolbar>
<button rcButtonIcon><rc-material-icon>menu</rc-material-icon></button>
Design System
<span style="flex: 1 1 auto">&nbsp;</span>
<button rcButtonIcon><rc-material-icon>favorite</rc-material-icon></button>
<button rcButtonIcon><rc-material-icon>settings</rc-material-icon></button>
</rc-toolbar>`,
});

export const BothTextAndButton = templateForBothTextAndButton.bind({});
