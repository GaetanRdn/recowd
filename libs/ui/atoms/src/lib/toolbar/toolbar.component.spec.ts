import { Component } from '@angular/core';
import { TestBed, waitForAsync } from '@angular/core/testing';
import { ToolbarComponent } from './toolbar.component';
import { TemplateLookup } from '@recowd/test/utils';

describe('ToolbarComponent', () => {
  beforeEach(waitForAsync(() =>
    TestBed.configureTestingModule({
      declarations: [
        ToolbarComponent,
        OnlyTextHostComponent,
        WithButtonHostComponent,
      ],
    }).compileComponents()));

  it('should create with only text', () => {
    const templateLookup: TemplateLookup<OnlyTextHostComponent> =
      new TemplateLookup(OnlyTextHostComponent);
    templateLookup.detectChanges();
    expect(templateLookup.firstChildElement).toMatchSnapshot();
  });

  it('should create with button', () => {
    const templateLookup: TemplateLookup<WithButtonHostComponent> =
      new TemplateLookup(WithButtonHostComponent);
    templateLookup.detectChanges();
    expect(templateLookup.firstChildElement).toMatchSnapshot();
  });
});

@Component({
  template: ` <rc-toolbar>Design System</rc-toolbar>`,
})
class OnlyTextHostComponent {}

@Component({
  template: ` <rc-toolbar>
    <button>Action1</button>
  </rc-toolbar>`,
})
class WithButtonHostComponent {}
