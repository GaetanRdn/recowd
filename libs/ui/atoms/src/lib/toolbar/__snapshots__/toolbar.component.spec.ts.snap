// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarComponent should create with button 1`] = `
<rc-toolbar>
  <button>
    Action1
  </button>
</rc-toolbar>
`;

exports[`ToolbarComponent should create with only text 1`] = `
<rc-toolbar>
  Design System
</rc-toolbar>
`;
